qiskit_ibm_experiment/service/ibm_experiment_service.py

```diff
@@ -937,7 +942,10 @@ def analysis_result(
         with map_api_error(f"Analysis result {result_id} not found."):
             raw_data = self._api_client.analysis_result_get(result_id)
 
-        return self._api_to_analysis_result(json.loads(raw_data, cls=json_decoder))
```

Comment:
Maybe just make `self._api_to_analysis_results()` return `AnalyasisResultData` instead of a dict?